birdnet2sqlite.py

```diff
@@ -2,40 +2,26 @@
 
 import argparse
 import ast
-import csv
 import datetime
 import re
 
 import sqlite_utils
 
-recorder_filename_date = re.compile(r"\d{8}_\d{6}")
-
-
-def parse_tsv(fp):
-    return csv.DictReader(fp, delimiter="\t")
-
-
-def autocast(obj):
-    for row in obj:
-        for key, value in row.items():
-            try:
-                row[key] = ast.literal_eval(value)
-            except (SyntaxError, ValueError):
-                pass
-        yield row
+from utils import parse_tsv, autocast
+from preprocess_birdnet_result import add_info
 
+recorder_filename_date = re.compile(r"\d{8}_\d{6}.BirdNET.selection.table.txt")
 
 def filename_to_datetime(filename):
     matches = recorder_filename_date.search(filename)
     if not matches:
         return  # Invalid filename
     try:
-        dt = datetime.datetime.strptime(matches.group(0), "%Y%m%d_%H%M%S")
+        dt = datetime.datetime.strptime(matches.group(0), "%Y%m%d_%H%M%S.BirdNET.selection.table.txt")
```

Comment:
You can use `$` at the end of your regex to be sure you are matching the end of your filename.